com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/ActivitiExceptionHandler.java

```diff
@@ -0,0 +1,77 @@
+package com.sap.cloud.lm.sl.cf.process.util;
+
+import java.sql.Timestamp;
+import java.text.MessageFormat;
+
+import org.activiti.engine.delegate.event.ActivitiEvent;
+import org.activiti.engine.delegate.event.ActivitiExceptionEvent;
+import org.activiti.engine.history.HistoricVariableInstance;
+import org.activiti.engine.impl.persistence.entity.VariableInstance;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.sap.cloud.lm.sl.common.SLException;
+import com.sap.cloud.lm.sl.persistence.model.ProgressMessage;
+import com.sap.cloud.lm.sl.persistence.model.ProgressMessage.ProgressMessageType;
+import com.sap.cloud.lm.sl.persistence.services.ProgressMessageService;
+
+public class ActivitiExceptionHandler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ActivitiExceptionHandler.class);
+
+    private static final String EXCEPTION_OCCURED_ERROR_MSG = "Exception occured during execution: \"{0}\"";
+
+    private ProgressMessageService progressMessageService;
+
+    private ActivitiEvent event;
+
+    public ActivitiExceptionHandler(ActivitiEvent event, ProgressMessageService progressMessageService) {
+        this.event = event;
+        this.progressMessageService = progressMessageService;
+    }
+
+    public void handle() {
+        ActivitiExceptionEvent activitiExceptionEvent = (ActivitiExceptionEvent) event;
+        String activitiExceptionMsg = activitiExceptionEvent.getCause().getMessage();
+
+        if (activitiExceptionMsg == null) {
+            return;
+        }
+
+        VariableInstance variableInstance = event.getEngineServices()
+            .getRuntimeService()
+            .getVariableInstance(event.getExecutionId(), com.sap.cloud.lm.sl.persistence.message.Constants.INDEXED_STEP_NAME);
+        String taskId = null;
+
+        if (variableInstance == null) {

```

Comment:
Extract this to another method and remove the temp variable taskId :) Also put the variableInstance null check in the method => this will make more clear caller method.